// testrail-js-api/lib/cases.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteCases = exports.deleteCase = exports.updateCases = exports.updateCase = exports.addCase = exports.getHistoryForCase = exports.getCases = exports.getCase = void 0;
function getCase(case_id) {
    return this.apiGet("get_case/" + case_id);
}
exports.getCase = getCase;
function getCases(project_id, queryVariables) {
    return this.apiGet("get_cases/" + project_id, {
        queryVariables,
    });
}
exports.getCases = getCases;
function getHistoryForCase(case_id) {
    return this.apiGet("get_history_for_case/" + case_id);
}
exports.getHistoryForCase = getHistoryForCase;
function addCase(section_id, case_data) {
    return this.apiPost("add_case/" + section_id, case_data);
}
exports.addCase = addCase;
function updateCase(case_id, case_data) {
    return this.apiPost("update_case/" + case_id, case_data);
}
exports.updateCase = updateCase;
function updateCases(project_id, case_data, queryVariables) {
    return this.apiPost("update_cases/" + project_id, case_data, {
        queryVariables,
    });
}
exports.updateCases = updateCases;
function deleteCase(case_id) {
    return this.apiPost("delete_case/" + case_id);
}
exports.deleteCase = deleteCase;
function deleteCases(project_id, case_ids, queryVariables) {
    return this.apiPost("delete_cases/" + project_id, { case_ids }, { queryVariables });
}
exports.deleteCases = deleteCases;
