// testrail-js-api/lib/configurations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteConfig = exports.deleteConfigGroup = exports.updateConfig = exports.updateConfigGroup = exports.addConfig = exports.addConfigGroup = exports.getConfigs = void 0;
function getConfigs(project_id) {
    return this.apiGet("get_configs/" + project_id);
}
exports.getConfigs = getConfigs;
function addConfigGroup(project_id, configuration_group) {
    return this.apiPost("add_config_group/" + project_id, configuration_group);
}
exports.addConfigGroup = addConfigGroup;
function addConfig(config_group_id, config) {
    return this.apiPost("add_config/" + config_group_id, config);
}
exports.addConfig = addConfig;
function updateConfigGroup(config_group_id, config) {
    return this.apiPost("update_config_group/" + config_group_id, config);
}
exports.updateConfigGroup = updateConfigGroup;
function updateConfig(config_id, config) {
    return this.apiPost("update_config/" + config_id, config);
}
exports.updateConfig = updateConfig;
function deleteConfigGroup(config_group_id) {
    return this.apiPost("delete_config_group/" + config_group_id);
}
exports.deleteConfigGroup = deleteConfigGroup;
function deleteConfig(config_id) {
    return this.apiPost("delete_config/" + config_id);
}
exports.deleteConfig = deleteConfig;
