// testrail-js-api/lib/plans.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteRunFromPlanEntry = exports.deletePlanEntry = exports.deletePlan = exports.closePlan = exports.updateRunInPlanEntry = exports.updatePlanEntry = exports.updatePlan = exports.addRunToPlanEntry = exports.addPlanEntry = exports.addPlan = exports.getPlans = exports.getPlan = void 0;
function getPlan(plan_id) {
    return this.apiGet("get_plan/" + plan_id);
}
exports.getPlan = getPlan;
function getPlans(project_id, filters) {
    return this.apiGet("get_plans/" + project_id, {
        queryVariables: filters,
    });
}
exports.getPlans = getPlans;
function addPlan(project_id, plan) {
    return this.apiPost("add_plan/" + project_id, plan);
}
exports.addPlan = addPlan;
function addPlanEntry(plan_id, data) {
    return this.apiPost("add_plan_entry/" + plan_id, data);
}
exports.addPlanEntry = addPlanEntry;
function addRunToPlanEntry(plan_id, entry_id, data) {
    return this.apiPost("add_run_to_plan_entry/" + plan_id + "/" + entry_id, data);
}
exports.addRunToPlanEntry = addRunToPlanEntry;
function updatePlan(plan_id, data) {
    return this.apiPost("update_plan/" + plan_id, data);
}
exports.updatePlan = updatePlan;
function updatePlanEntry(plan_id, entry_id, data) {
    return this.apiPost("update_plan_entry/" + plan_id + "/" + entry_id, data);
}
exports.updatePlanEntry = updatePlanEntry;
function updateRunInPlanEntry(run_id, data) {
    return this.apiPost("update_run_in_plan_entry/" + run_id, data);
}
exports.updateRunInPlanEntry = updateRunInPlanEntry;
function closePlan(plan_id) {
    return this.apiPost("close_plan/" + plan_id);
}
exports.closePlan = closePlan;
function deletePlan(plan_id) {
    return this.apiPost("delete_plan/" + plan_id);
}
exports.deletePlan = deletePlan;
function deletePlanEntry(plan_id, entry_id) {
    return this.apiPost("delete_plan_entry/" + plan_id + "/" + entry_id);
}
exports.deletePlanEntry = deletePlanEntry;
function deleteRunFromPlanEntry(run_id) {
    return this.apiPost("delete_run_from_plan_entry/" + run_id);
}
exports.deleteRunFromPlanEntry = deleteRunFromPlanEntry;
