// testrail-js-api/lib/results.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addResultsForCases = exports.addResults = exports.addResultForCase = exports.addResult = exports.getResultsForRun = exports.getResultsForCase = exports.getResults = void 0;
function getResults(test_id, filters) {
    return this.apiGet("get_results/" + test_id, {
        queryVariables: filters,
    });
}
exports.getResults = getResults;
function getResultsForCase(run_id, case_id, filters) {
    return this.apiGet("get_results_for_case/" + run_id + "/" + case_id, { queryVariables: filters });
}
exports.getResultsForCase = getResultsForCase;
function getResultsForRun(run_id, filters) {
    return this.apiGet("get_results_for_run/" + run_id, {
        queryVariables: filters,
    });
}
exports.getResultsForRun = getResultsForRun;
function addResult(test_id, data) {
    return this.apiPost("add_result/" + test_id, data);
}
exports.addResult = addResult;
function addResultForCase(run_id, case_id, data) {
    return this.apiPost("add_result_for_case/" + run_id + "/" + case_id, data);
}
exports.addResultForCase = addResultForCase;
function addResults(run_id, results) {
    return this.apiPost("add_results/" + run_id, {
        results,
    });
}
exports.addResults = addResults;
function addResultsForCases(run_id, results) {
    return this.apiPost("add_results_for_cases/" + run_id, {
        results,
    });
}
exports.addResultsForCases = addResultsForCases;
